File: app/src/main/java/com/example/easyhttp/http/exception/ApiException.java
package com.example.easyhttp.http.exception;

public class ApiException extends RuntimeException {
    public static final int Code_TimeOut = 1000;
    public static final int Code_UnConnected = 1001;
    public static final int Code_MalformedJson = 1020;
    public static final int Code_Default = 1003;
    public static final String CONNECT_EXCEPTION = "网络连接异常，请检查您的网络状态";
    public static final String SOCKET_TIMEOUT_EXCEPTION = "网络连接超时，请检查您的网络状态，稍后重试";
    public static final String MALFORMED_JSON_EXCEPTION = "数据解析错误";

    public ApiException(int resultCode, String msg) {
        this(getApiExceptionMessage(resultCode, msg));
    }

    public ApiException(String detailMessage) {
        super(detailMessage);
    }

    /**
     * @param code
     * @return
     */
    private static String getApiExceptionMessage(int code, String msg) {
        String message;
        switch (code) {
            default:
                message = code + "#" + msg;
        }
        return message;
    }
}

File: app/src/main/java/com/example/easyhttp/http/exception/ApiException.kt
package com.example.easyhttp.http.exception

import java.lang.RuntimeException

/**
 * Created by shiwei on 2018/1/12.
 */
class ApiException(resultCode: Int, msg: String) : RuntimeException(getApiExceptionMessage(resultCode, msg)) {
    companion object {
        val Code_TimeOut = 1000
        val Code_UnConnected = 1001
        val Code_MalformedJson = 1020
        val Code_Default = 1003
        val CONNECT_EXCEPTION = "网络出现了些许小问题,请检查网络连接"
        val SOCKET_TIMEOUT_EXCEPTION = "网络连接超时，请检查您的网络状态，稍后重试"
        val MALFORMED_JSON_EXCEPTION = "数据解析错误"
    }

    init {
        var result = getApiExceptionMessage(resultCode, msg)
    }
}

fun getApiExceptionMessage(resultCode: Int, msg: String): String {
    val message: String
    when (resultCode) {
        else -> message = resultCode.toString() + "#" + msg
    }
    return message
}
